`hyperband` and `rl` search strategies are nut running for Tabular prediction.

```
import autogluon as ag
from autogluon.task import TabularPrediction as task

import sklearn.datasets
import pandas as pd

cancer_data = sklearn.datasets.load_breast_cancer(return_X_y=False)
data = pd.DataFrame(cancer_data['data'], columns=cancer_data['feature_names'])
data['target'] = cancer_data['target']

gbm_options = {
    'num_boost_round': 100,
    'num_leaves': ag.space.Int(lower=26, upper=66, default=36),
}

hyperparameters = {
    'GBM': gbm_options,
}

hp_tune = True
time_limits = 2 * 60
num_trials = 3
output_directory = './agModels--tuningCancerDataset'
data = task.Dataset(data)
label_column = 'target'
search_strategy = 'hyperband'
predictor = task.fit(
    train_data=data,
    holdout_frac=0.3,
    label=label_column,
    output_directory=output_directory,
    time_limits=time_limits,
    num_trials=num_trials,
    hyperparameter_tune=hp_tune,
    hyperparameters=hyperparameters,
    search_strategy=search_strategy,
    auto_stack=False,
    stack_ensemble_levels=0
)
```

The error: 
```
File "/Users/rmelikbe/Desktop/repos/autogluon/autogluon/utils/tabular/ml/trainer/abstract_trainer.py", line 354, in train_single_full
    Y_train=y_train, Y_test=y_test, scheduler_options=(self.scheduler_func, self.scheduler_options), verbosity=self.verbosity)
  File "/Users/rmelikbe/Desktop/repos/autogluon/autogluon/utils/tabular/ml/models/lgb/lgb_model.py", line 285, in hyperparameter_tune
    scheduler = scheduler_func(lgb_trial, **scheduler_options)
  File "/Users/rmelikbe/Desktop/repos/autogluon/autogluon/scheduler/hyperband.py", line 136, in __init__
    reward_attr=reward_attr, visualizer=visualizer, dist_ip_addrs=dist_ip_addrs)
  File "/Users/rmelikbe/Desktop/repos/autogluon/autogluon/scheduler/fifo.py", line 91, in __init__
    self.searcher = searcher_factory(searcher, **kwargs)
  File "/Users/rmelikbe/Desktop/repos/autogluon/autogluon/searcher/searcher_factory.py", line 14, in searcher_factory
    raise AssertionError("name = '{}' not supported".format(name))
AssertionError: name = 'hyperband' not supported
```

--- a/autogluon/task/tabular_prediction/tabular_prediction.py
+++ b/autogluon/task/tabular_prediction/tabular_prediction.py
@@ -272,7 +272,7 @@ class TabularPrediction(BaseTask):
             dist_ip_addrs = []
 
         if search_options is None:
-            search_options = {}
+            search_options = dict()
 
         if hyperparameters is None:
             hyperparameters = {
@@ -355,7 +355,15 @@ class TabularPrediction(BaseTask):
         }
         if isinstance(search_strategy, str):
             scheduler = schedulers[search_strategy.lower()]
+            # This is a fix for now. But we need to separate between scheduler
+            # (mainly FIFO and Hyperband) and searcher. Currently, most searchers
+            # only work with FIFO, and Hyperband works only with random searcher,
+            # but this will be different in the future.
+            if search_strategy == 'hyperband':
+                # Currently, HyperbandScheduler only supports random searcher
+                scheduler_options['searcher'] = 'random'
         else:
+            # TODO: Check that search_strategy is a subclass of TaskScheduler
             assert callable(search_strategy)
             scheduler = search_strategy
             scheduler_options['searcher'] = 'random'

